### PR TITLE
Merge :class:`~.OpenGLMobject` and :class:`~.Mobject`

### PR DESCRIPTION
**Pull request by TRoboto**
## Overview / Explanation for Changes
Many functions are shared between OpenGLMobject and Mobject, and also most of the OpenGL classes are just redundant. Therefore, I am trying to combine them into one Mobject class. For now, the code looks really messy, but it works. Even [this bug](https://github.com/ManimCommunity/manim/issues/1162#issue-839201943) has been implicitly resolved!

SurfaceExample still doesn't run as expected, and I really would love to have some help here. All comments are most welcome. 

## Testing Status

None :/

## Further Comments

VMobject and OpenGLVMobject still need to be combined when we finish here.

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have chosen a descriptive PR title (see top of PR template for examples)

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The PR title is descriptive enough

### CODE REVIEW

**Comment by TRoboto** [comment]
All examples work now. Waiting for your feedback!

**Comment by MrMallIronmaker** [comment]
It looks like there was a regression in the ImageMobject class. Are you able to run the tests locally, `git bisect` for when it was introduced, and fix the bug?

**Comment by TRoboto** [comment]
All tests pass now and all OpenGL scenes work perfectly. I think this PR is ready to be merged.

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -997,35 +1117,39 @@ def scale(self, scale_factor: float, **kwargs) -> "Mobject":
...
-            return self
         else:
             self.apply_points_function_about_point(
                 lambda points: scale_factor * points, **kwargs
             )
-            return self
-
-    def rotate_about_origin(self, angle, axis=OUT, axes=[]):
+        return self
+        # self.apply_function_to_points(
```
Comment (MrMallIronmaker) [inline manim/mobject/mobject.py:1131]: Again - just some commented code to be removed

**Review by MrMallIronmaker — CHANGES_REQUESTED** [review]
This is a huge change (lines of code wise) but I'm OK playing fast and loose with the opengl functionality because it's experimental. If you've run the examples and they're fine, and you've also resolved another bug, then I think this PR is definitely a (big!) step in the right direction.

My main concern is deleting all the commented code, with a non-blocking question about the high-level purpose of OpenGLMobject is now.

File: manim/mobject/mobject.py (hunk 1 above)
Comment (TRoboto) [reply manim/mobject/mobject.py]: Thank you for your feedback. OpenGLMobject has been removed along with all the comments in the Mobject class.

**Comment by TRoboto** [comment]
Thank you for your feedback. OpenGLMobject has been removed along with all the comments in the Mobject class.

File: manim/mobject/mobject.py (hunk 1 above)
Comment (TRoboto) [reply manim/mobject/mobject.py]: Removed.

File: manim/mobject/opengl_mobject.py [hunk 2]
```diff
@@ -60,256 +47,250 @@ def __init__(
...
 
-        self.shader_indices = None
+        # self.shader_indices = None
 
-        if self.depth_test:
-            self.apply_depth_test()
+        # if self.depth_test:
+        #     self.apply_depth_test()
 
     def __str__(self):
```
[the remark this answers is not in the document]
Comment (TRoboto) [reply manim/mobject/opengl_mobject.py]: An OpenGL object is no longer needed and has been removed.

File: manim/mobject/opengl_mobject.py [hunk 3]
```diff
@@ -60,256 +47,250 @@ def __init__(
...
-        self.submobjects = []
-        self.parents = []
-        self.family = [self]
-        self.locked_data_keys = set()
-        self.needs_new_bounding_box = True
-
-        self.init_data()
-        self.init_uniforms()
-        self.init_updaters()
+        # self.opacity = opacity
```
[the remark this answers is not in the document]
Comment (TRoboto) [reply manim/mobject/opengl_mobject.py]: Deleted.

**Comment by MrMallIronmaker** [comment]
I take it that OpenGLMobject didn't have any important roles then?

**Comment by TRoboto** [comment]
> I take it that OpenGLMobject didn't have any important roles then?

Yes, OpenGLMobject shares many functions with the regular Mobject. Even things listed under "self.opengl" now may be removed in the future, I wasn't 100% comfortable with removing them at this point. Once the tests are ready, I believe it will be the best time to really refactor Mobject.

I think the reason why there is a separate Mobject for OpenGL is just the need to quickly carry it over to the community version. 

One thing, I am not sure how IntelliSense would work if there are functions that can only work with Cairo/OpenGL (users won't know until they play the scene).

Please note that I also removed `_AnimationBuilder` and `OpenGLGroup` classes.

**Review by MrMallIronmaker — APPROVED** [review]
Looks good to me!

**Comment by eulertour** [comment]
This change breaks a lot of my work on porting Latex and Text to OpenGL, and the errors it caused are difficult to debug due to the interleaving of the inheritance chains. I think it'd be best to revert this until all of the OpenGL mobjects are working.

**Pull request merged**